### PR TITLE
Update to `rodio` 0.18

### PR DESCRIPTION
**Pull request by BD103**
# Objective

- Adopted from #13528.
- `rodio` released 0.18! While we are working on migrating away from it and towards `kira`, it is still good to keep our dependencies up-to-date.

## Solution

- Update `Cargo.toml` to depend on `rodio` 0.18.
  - #13528 was failing because it didn't update `rodio` for `wasm32-unknown-unknown` too.

## Testing

- The CI should catch any errors here, but you can also run an audio example if you want like `spatial_audio_2d`.

---

## Changelog

- Updated `rodio` to 0.18.

### CODE REVIEW

**Review by Olle-Lukowski — APPROVED** [review]
Simple change, assuming the examples still work (can't check right now), this looks fine!

**Comment by alice-i-cecile** [comment]
Release notes: https://github.com/RustAudio/rodio/blob/master/CHANGELOG.md#version-0180-2024-05-05

**Review by SolarLiner — APPROVED** [review]
LGTM, also assuming the examples still work, and having looked at the rodio changelog

**Comment by mweatherley** [comment]
I don't know how keen to actually go forward with it I am, but I'll mention that rodio 0.18 includes the change I made to `SpatialSink`, which theoretically allows the current audio system to be restructured to reuse its audio sinks.

**Review by mnmaita — APPROVED** [review]

**Comment by BD103** [comment]
> I don't know how keen to actually go forward with it I am, but I'll mention that rodio 0.18 includes the change I made to `SpatialSink`, which theoretically allows the current audio system to be restructured to reuse its audio sinks.

If / once this gets merged, I'll create an issue for this and check if the Better Audio working group is interested. They can decide whether it's worth working on, or if all efforts should be put into the `kira` migration instead.

**Review by mockersf — APPROVED** [review]

**Pull request merged**
